### PR TITLE
Update layout

### PR DESCRIPTION
**Pull request by markknol**
## Motivation
I think the nekovm.org website looks outdated. I can imaging it is not an issue for all actual users, but I think it is better to suggest the project is still alive with an nice website. Since it is somehow related to Haxe I think it should look nicer than it looks now. I browsed through the current site and noticed the navigation in the docs was not really clear, while there is quite some useful information available.

## Changes
- a new skin with more clear navigation
- an updated homepage.
- I renamed news to releases.
- I updated some pages to make the navigation more clear
- The specs page is split into separate pages.

#### ufront thingies
- I did some things to enable sub navigation in the sidemenu.
- I disabled the client part of ufront. I had problems with a ufront client pushstate something  and it causes issues with the sidemenu. It saves ~450kb of JavaScript.

> I have no idea what I'm doing when it comes to ufront

## Preview

I'm not sure if we have a staging for nekovm.org, but the change looks like this:

neko-vm

cc @ncannasse

### CODE REVIEW

**Comment by nadako** [comment]
Looks nice!

**Comment by Simn** [comment]
Looks very markish, which is a good thing!

**Comment by fullofcaffeine** [comment]
Nice!

File: src/NekoSite.hx [hunk 1]
```diff
@@ -67,21 +83,36 @@ class NekoSite extends Controller {
...
+			return pvr;
+		};
+	}
+	
 	@:route("/*")
 	public function page( rest:Array<String> ) {
 		var pageName = (rest.length>0) ? rest.join("/") : "index";
-		var pvr = new PartialViewResult({});
+		var pvr = new PartialViewResult({}, pageName == "index" ? "index.html" : null);
 		return nekoApi.getPage( pageName ) >> function( page:Page ):ViewResult {
```
Comment (fullofcaffeine) [inline src/NekoSite.hx]: What's `>>` for?

Comment (fullofcaffeine) [inline src/NekoSite.hx]: Btw, does it handle 404 if the markdown is not found?

**Comment by ncannasse** [comment]
Ok for me :)

File: src/NekoSite.hx (hunk 1 above)
Comment (kevinresol) [inline src/NekoSite.hx]: > What's >> for? I don't recognise this syntax.

It maps a future to another value. See https://github.com/haxetink/tink_core#operators

**Comment by kevinresol** [comment]
@markknol do you have any specific ufront questions in mind? if yes I could probably add an example in https://github.com/kevinresol/ufront-nodejs-guide/

File: src/NekoSite.hx (hunk 1 above)
Comment (fullofcaffeine) [inline src/NekoSite.hx]: Nice!

Comment (fullofcaffeine) [inline src/NekoSite.hx]: Thanks, Kevin.

**Comment by Simn** [comment]
For some reason this cannot be merged, could send another PR?

**Comment by ibilon** [comment]
Can't be merged because it's not branching of master/

**Comment by Simn** [comment]
> Can't be merged because it's not branching of master.

I'm not even sure what that means... Anyway, someone fix it please!

**Comment by Merelleya** [comment]
@markknol this looks absolutely gorgeous!

You are like a never-ending supply of "oh this is so pretty and makes me happy"-candy. whenever I see you work, I really just sit in front of the screen and enjoy the happy "oh pretty" for a few minutes.

Thank you so much!

**Comment by nadako** [comment]
I love those gifs :)

**Comment by markknol** [comment]
@Simn I dont understand why it cannot be merged? What should I do now? The code is still in the `new_layout` branch, do not delete that please.

**Comment by markknol** [comment]
@simn hey now it works, whats the next step?

**Comment by Simn** [comment]
Uhm, I was thinking that I press that merge button now and then the page looks good. Is that not the case?

**Comment by ibilon** [comment]
The travis file has deploy instructions so it should be enough.

**Comment by Simn** [comment]
Well then let's do it!

**Comment by ibilon** [comment]
Worked http://nekovm.org/

**Comment by Simn** [comment]
NIce, that was easy!

**Comment by markknol** [comment]
Ha wow that was fast. 

Only the 404 needs to be done I guess.

**Comment by markknol** [comment]
@merelleya do we track stats for this site?

**Comment by ibilon** [comment]
You can make one with this https://github.com/HaxeFoundation/haxe.org/blob/staging/src/Server.hx#L27-L28
and add it like this https://github.com/HaxeFoundation/haxe.org/blob/staging/src/Server.hx#L34

**Comment by markknol** [comment]
Ok well it has no prio since it wasnt designed yet.

If anyone else feels the need to spice it up a bit, please do.

**Comment by Merelleya** [comment]
Hey @markknol 

We are not tracking this at the moment. I will set up a Tag Manager container and send you the details.

**Pull request merged**